app/serializers/concerns/optional_serialized_attributes.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+module OptionalSerializedAttributes
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def attribute_if_positive(optional_attribute)
+      attribute optional_attribute, if: -> { object.send(optional_attribute).positive? }
+    end
```

Comment:
It's not just a matter of bloated responses. 

1. You have a schema that acts as a form of documentation for what the object describes. This schema should be consistent so that data consumer knows how to build an object and what attribute will be present. Having attributes that are sometimes present, sometimes not complicates this process.

2. There's a debugging aspect too, which is you don't know if the property is ommitted due to a bug or processing error vs intentionally being left off.

3. Almost all Public Facing APIs will provide null value keys. There are many reasons why this is useful to a consumer of a public facing API that goes beyond the benefits to be gained by marginally less bloated responses.

If anything, JSONAPI allows for a consumer of the data to specify what attributes they want returned. If we want to mitigate bloated responses we should follow that approach.